File: 1726-tuple-with-same-product/1726-tuple-with-same-product.java
class Solution {
    public int tupleSameProduct(int[] nums) {
        int ans = 0;
        HashMap<Integer,Integer> hm = new HashMap<>();
        
        for(int i = 0;i<nums.length;i++){
            for(int j = i+1;j<nums.length;j++){
                int prod = nums[i] * nums[j];
                hm.put(prod,hm.getOrDefault(prod,0)+1);
            }
        }
        for(int val : hm.values()){
            int fact = ((val*(val-1))/2)*8;
            ans += fact;
        }
        return ans;
    }
}

File: 1726-tuple-with-same-product/1726-tuple-with-same-product.cpp
class Solution{
public:
    int tupleSameProduct(vector<int>& nums){
        int n = nums.size();
        unordered_map<int,int> mp;
        
        for(int i = 0;i<n;i++){
            for(int j = i+1;j<n;j++){
                int prod = nums[i]*nums[j];
                mp[prod]++;
            }
        }
        int ans = 0;
        for(auto&it:mp){
            ans += ((it.second)*(it.second-1))/2;
        }
        
        return ans*8;
    }
};